Neos.Flow/Classes/Configuration/ConfigurationManager.php

```diff
@@ -232,70 +197,44 @@ public function setPackages(array $packages)
      */
     public function getAvailableConfigurationTypes(): array
     {
-        return array_keys($this->configurationTypes);
+        return array_keys($this->configurationLoaders);
     }
 
     /**
-     * Resolve the processing type for the configuration type.
-     *
-     * This returns the CONFIGURATION_PROCESSING_TYPE_* to use for the given
-     * $configurationType.
+     * Registers a new configuration type with the given source.
      *
-     * @param string $configurationType
-     * @return string
-     * @throws Exception\InvalidConfigurationTypeException on non-existing configurationType
+     * @param string $configurationType The type to register, may be anything
+     * @param LoaderInterface|callable|string $configurationLoader
+     * @return void
+     *@throws \InvalidArgumentException on invalid configuration processing type
      */
-    public function resolveConfigurationProcessingType(string $configurationType): string
+    public function registerConfigurationType(string $configurationType, $configurationLoader): void
```

Comment:
This means we need to target the next major, strictly speaking. I see two options:

- **Provide a code migration** that helps migrate or at least spot code that needs to be adjusted.
- **Make the `$configurationLoader` optional (as long as it may be a string for B/C reasons) and fill it depending on the type in `convertLegacyProcessingType()`

Reply:
The signature has changed, but it should only break if someone actually *subclassed* the `ConfigurationManager` (that should have been `final` IMO).
The *implementation* is mostly backwards compatible, i.e.
```php
$configurationManager->registerConfigurationType('SomeType');
$configurationManager->registerConfigurationType('SomeType', ConfigurationManager::CONFIGURATION_PROCESSING_TYPE_OBJECTS);
$configurationManager->registerConfigurationType('SomeType', ConfigurationManager::CONFIGURATION_PROCESSING_TYPE_OBJECTS, true);
```
should all still work (thanks to line 215) with the exception that the last param (`allowSplitSource`) is ignored since that now just depends on the loader.

I think that both (overriding ConfigurationManager and relying on "allowSplitSource" to be false(!)) could be considered edge cases and don't require a migration nor major version jump as long as the "breaking change" is properly described (which it _is_ in this PR IMO)

Reply:
But `$configurationManager->registerConfigurationType('SomeType');` will lack the **required** second parameter!

Reply:
*dang* of course.. That's why we need reviews :)
What about
```suggestion
    public function registerConfigurationType(string $configurationType, $configurationLoader = null): void
```
and sth along the lines of
```php
if ($configurationLoader === null) {
    $configurationLoader = new SettingsLoader(new YamlSource());
// B/C layer
} elseif (is_string($configurationLoader)) {
    $configurationLoader = $this->convertLegacyProcessingType($configurationType, $configurationLoader);
}
```

in the implementation?

Reply:
```php
$configurationLoader = new MergeLoader(new YamlSource(), $configurationType);
```
not
```php
$configurationLoader = new SettingsLoader(new YamlSource());
```
of course